tcms/rpc/tests/test_testexecution.py

```diff
@@ -355,25 +355,64 @@ def test_update_with_single_caserun(self):
         self.assertEqual(execution['sortkey'], 90)
 
     def test_update_with_assignee_id(self):
-        self.assertNotEqual(self.case_run_1.assignee, self.user.pk)
+        self.assertNotEqual(self.case_run_1.assignee, self.user)
         execution = self.rpc_client.TestExecution.update(self.case_run_1.pk, {
             "assignee": self.user.pk
         })
+        self.case_run_1.refresh_from_db()
+
         self.assertEqual(execution['assignee'], self.user.username)
+        self.assertEqual(self.case_run_1.assignee, self.user)
 
     def test_update_with_assignee_email(self):
-        self.assertNotEqual(self.case_run_1.assignee, self.user.email)
+        self.assertNotEqual(self.case_run_1.assignee, self.user)
         execution = self.rpc_client.TestExecution.update(self.case_run_1.pk, {
             "assignee": self.user.email
         })
+        self.case_run_1.refresh_from_db()
+
         self.assertEqual(execution['assignee'], self.user.username)
+        self.assertEqual(self.case_run_1.assignee, self.user)
 
     def test_update_with_assignee_username(self):
-        self.assertNotEqual(self.case_run_1.assignee, self.user.username)
+        self.assertNotEqual(self.case_run_1.assignee, self.user)
         execution = self.rpc_client.TestExecution.update(self.case_run_1.pk, {
             "assignee": self.user.username
         })
+        self.case_run_1.refresh_from_db()
+
         self.assertEqual(execution['assignee'], self.user.username)
+        self.assertEqual(self.case_run_1.assignee, self.user)
+
+    def test_update_with_tested_by_id(self):
+        self.assertNotEqual(self.case_run_2.tested_by, self.user)
+        execution = self.rpc_client.TestExecution.update(self.case_run_2.pk, {
+            "tested_by": self.user.pk
+        })
+        self.case_run_2.refresh_from_db()
+
+        self.assertEqual(execution['tested_by'], self.user.username)
+        self.assertEqual(self.case_run_2.tested_by, self.user)
+
+    def test_update_with_tested_by_email(self):
+        self.assertNotEqual(self.case_run_2.tested_by, self.user)
+        execution = self.rpc_client.TestExecution.update(self.case_run_2.pk, {
+            "tested_by": self.user.email
+        })
+        self.case_run_2.refresh_from_db()
+
+        self.assertEqual(execution['tested_by'], self.user.username)
+        self.assertEqual(self.case_run_2.tested_by, self.user)
+
+    def test_update_with_tested_by_username(self):
+        self.assertNotEqual(self.case_run_2.tested_by, self.user)
+        execution = self.rpc_client.TestExecution.update(self.case_run_2.pk, {
+            "tested_by": self.user.username
+        })
+        self.case_run_2.refresh_from_db()
+
+        self.assertEqual(execution['tested_by'], self.user.username)
+        self.assertEqual(self.case_run_2.tested_by, self.user)
```

Comment:
FYI you could have merged these scenarios with the already existing ones and update/assert the 2 fields at the same time. This way also works.